extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts

```diff
@@ -0,0 +1,295 @@
+            else { continue; }
+            InstrumentFunc(target, tracker, {
+              req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
+              rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
+            });
+
+            debugBins[tracker] = new DebugBin(tracker, 0, debugBinParent, (i + 1) * 50);
+          }
+
+          // 6. append permanent style
+          permStyleEl = document.createElement("style");
+          permStyleEl.innerHTML = permStyle;
+          document.head.appendChild(permStyleEl);
+
+          // 7. setup temporary style and root element
+          tempStyleEl = document.createElement("style");
+          tempStyleEl.innerHTML = tempStyle;
+          const rootEl = rootElement = document.createElement("div");
+          // TODO: research more accessibility (aria)
+          rootEl.style.position = 'fixed';
+          rootEl.style.width = '100vw';
+          rootEl.style.height = '100vh';
+          rootEl.style.backgroundColor = '#ffffff';
+          rootEl.style.opacity = '0';
+          rootEl.style.pointerEvents = 'none';
+          rootEl.style.top = '-100%';
+          rootEl.style.transition = '.2s top cubic-bezier(0.87, 0, 0.13, 1)';
+
+          // 8. add button to debugBinParent
+          debugBinParent.addButton((evt: MouseEvent) => {
+            evt.stopPropagation();
+            rootEl.style.top = (rootEl.style.opacity === '0') ? '0%' : '-100%';
+            rootEl.style.pointerEvents = (rootEl.style.opacity === '0') ? 'auto' : 'none';
+
+            if (rootEl.style.opacity === '0') {
+              document.head.appendChild(tempStyleEl);
+            } else {
+              document.head.removeChild(tempStyleEl);
+            }
+
+            rootEl.style.opacity = (rootEl.style.opacity === '0') ? '1' : '0';
+          }, 'toggle detailed view');
+
+          // 9. setup logger and log config
+          const logHeading = document.createElement("h1");
+          logHeading.textContent = 'detailed log';
+          logHeading.style.textAlign = 'center';
+          rootEl.appendChild(logHeading);
+
+          logger = new LoggingElement(rootEl);
+
+          document.body.appendChild(
+            rootEl
+          );
+
+          document.body.appendChild(
+            debugBinContainer
+          );
+
+          logger.newLogEntry(config, `[0s] config`, 0);
+        }
+      }
+
+      _self.preProcessItem = (itemType: string) => {
+        return (funcArgs: IInstrumentCallDetails, ...orgArgs: any[]) => {
+          (debugBins[itemType] || debugBins.default).increment();
+          if (itemType === 'trackException' && !debugBinParent.showChildren) {
+            debugBinParent.addClassToEl('notify');
+          }
+          logger.newLogEntry(funcArgs, `[${(+new Date() - startTime) / 1000}s] ${itemType}`, 0);
+          console.log(`[${itemType}] preProcess - funcArgs: `, funcArgs);
```

Comment:
we should check that both console and console.log are in fact not null (they can be on some browsers if the F12 is not open)